models/repo.go

```diff
@@ -1639,11 +1640,11 @@ func DeleteRepository(doer *User, uid, repoID int64) error {
 		return err
 	}
 
-	if err = sess.Commit(); err != nil {
+	if err = committer.Commit(); err != nil {
 		return err
 	}
 
-	sess.Close()
+	committer.Close()
```

Comment:
Any particular reason why it's already closed here? Their is a `defer commiter.Close()` in this function already going on.

Reply:
To release the connection ASAP because below code will begin another connection I think.